File: server/models/Comment.js
const mongoose = require('mongoose');

const commentSchema = new mongoose.Schema({
  author_username: { type: String, ref: 'User' },
  body:           { type: String },
  likes:          [{ type: String }],
  post_id_belong: { type: String }
});

module.exports = mongoose.model('Comment', commentSchema);

File: mern_mobile_app/lib/models/comment.dart
class Comment {
  final String id;
  final String authorUsername;
  final String body;
  final Set<String> likes;
  final String postIdBelong;
  final DateTime datePosted;

  Comment({
    required this.id,
    required this.authorUsername,
    required this.body,
    required this.likes,
    required this.postIdBelong,
    required this.datePosted
  });
}